runtime-decompiler/src/main/java/org/fife/ui/hex/swing/HexSearch.java

```diff
@@ -0,0 +1,160 @@
+package org.fife.ui.hex.swing;
+
+import java.util.ArrayList;
+
+public class HexSearch {
+
+    private HexEditor hex;
+    private int start = 0;
+    int end = 0;
+    int counter = 0;
+    boolean found = false;
+
+    public enum HexSearchOptions {
+        HEX,
+        INT,
+        TEXT
+    }
+
+    public HexSearch(HexEditor hex) {
+        this.hex = hex;
+    }
+
+    private ArrayList<Byte> getByteArray(String str, String type) {
+        ArrayList<Byte> arr = new ArrayList<>();
+        switch (type) {
+            case "TEXT":
+                byte[] bytesText = str.getBytes();
+                if (bytesText.length == 0) {
+                    return null;
+                }
+                for (byte b : bytesText) {
+                    arr.add(b);
+                }
+                break;
+
+            case "INT":
+                if (str.equals(" ")) {
+                    return null;
+                }
+                String[] spliced = str.split(" ");
+                for (String s : spliced) {
+                    try {
+                        arr.add(Integer.valueOf(s).byteValue());
+                    } catch (NumberFormatException e) {
+                        System.out.println("NAN");
+                        return null;
+                    }
+                }
+                break;
+
+            case "HEX":
+                if (str.equals("")) {
+                    return null;
+                }
+                String[] splicedHex = str.split(" ");
+                for (String s : splicedHex) {
+                    try {
+                        int i = Integer.parseInt(s, 16);
+                        arr.add(Integer.valueOf(i).byteValue());
+                    } catch (NumberFormatException e) {
+                        System.out.println("NAN");
+                        return null;
+                    }
+                }
+                break;
+
+            default:
+                return null;
```

Comment:
Instead of returning null, throwing some self describing  exception woudl be better

Reply:
I would throw also from here, in the defautl section

Btw, I think you shoud ldeclare your own exception to be thrown from this initialzation, as proepr reaction to it is necessary